crates/build-rs/src/input.rs

```diff
@@ -480,112 +445,158 @@ pub fn rustc_wrapper() -> Option<PathBuf> {
 }
 
 /// The home page from the manifest of your package.
 #[track_caller]
-pub fn cargo_pkg_homepage() -> String {
-    get_str("CARGO_PKG_HOMEPAGE")
+pub fn cargo_pkg_homepage() -> Option<String> {
+    to_opt(var_or_panic("CARGO_PKG_HOMEPAGE")).map(to_string)
 }
 
 /// The repository from the manifest of your package.
 #[track_caller]
-pub fn cargo_pkg_repository() -> String {
-    get_str("CARGO_PKG_REPOSITORY")
+pub fn cargo_pkg_repository() -> Option<String> {
+    to_opt(var_or_panic("CARGO_PKG_REPOSITORY")).map(to_string)
 }
 
 /// The license from the manifest of your package.
 #[track_caller]
-pub fn cargo_pkg_license() -> String {
-    get_str("CARGO_PKG_LICENSE")
+pub fn cargo_pkg_license() -> Option<String> {
+    to_opt(var_or_panic("CARGO_PKG_LICENSE")).map(to_string)
 }
 
 /// The license file from the manifest of your package.
 #[track_caller]
-pub fn cargo_pkg_license_file() -> PathBuf {
-    get_path("CARGO_PKG_LICENSE_FILE")
+pub fn cargo_pkg_license_file() -> Option<PathBuf> {
+    to_opt(var_or_panic("CARGO_PKG_LICENSE_FILE")).map(to_path)
 }
 
 /// The Rust version from the manifest of your package. Note that this is the
 /// minimum Rust version supported by the package, not the current Rust version.
 #[track_caller]
-pub fn cargo_pkg_rust_version() -> String {
-    get_str("CARGO_PKG_RUST_VERSION")
+pub fn cargo_pkg_rust_version() -> Option<String> {
+    to_opt(var_or_panic("CARGO_PKG_RUST_VERSION")).map(to_string)
 }
 
 /// Path to the README file of your package.
 #[track_caller]
-pub fn cargo_pkg_readme() -> PathBuf {
-    get_path("CARGO_PKG_README")
+pub fn cargo_pkg_readme() -> Option<PathBuf> {
+    to_opt(var_or_panic("CARGO_PKG_README")).map(to_path)
+}
+
+fn is_present(key: &str) -> bool {
+    var_os(key).is_some()
+}
+
+#[track_caller]
+fn var_or_panic(key: &str) -> std::ffi::OsString {
+    var_os(key).unwrap_or_else(|| panic!("cargo environment variable `{key}` is missing"))
+}
+
+fn to_path(value: std::ffi::OsString) -> PathBuf {
+    PathBuf::from(value)
+}
+
+#[track_caller]
+fn to_string(value: std::ffi::OsString) -> String {
+    match value.into_string() {
+        Ok(s) => s,
+        Err(value) => {
+            let err = std::str::from_utf8(value.as_encoded_bytes()).unwrap_err();
+            panic!("{err}")
+        }
+    }
+}
+
+fn to_opt(value: std::ffi::OsString) -> Option<std::ffi::OsString> {
+    (!value.is_empty()).then_some(value)
+}
+
+#[track_caller]
+fn to_strings(value: std::ffi::OsString, sep: char) -> Vec<String> {
+    if value.is_empty() {
+        return Vec::new();
+    }
+    let value = to_string(value);
+    value.split(sep).map(str::to_owned).collect()
+}
+
+#[track_caller]
+fn to_parsed<T>(value: std::ffi::OsString) -> T
+where
+    T: std::str::FromStr,
+    T::Err: std::fmt::Display,
+{
+    let value = to_string(value);
+    match value.parse() {
+        Ok(s) => s,
+        Err(err) => {
+            panic!("{err}")
+        }
+    }
```

Comment:
nit: any reason we can't do this?

```suggestion
    value.parse().unwrap()
```

Reply:
The question is whether to report Debug or Display